### PR TITLE
ITM 67 and 69

### PR DESCRIPTION
**Pull request by dereknop**

### CODE REVIEW

File: swagger_server/swagger/swagger.yaml [hunk 1]
```diff
@@ -1015,6 +1015,34 @@ components:
         - - treatment: Tourniquet
           - location: right forearm
         scenario_id: scenario_id
+    ProbeOption:
```
Comment (nextcen-dgemoets) [inline swagger_server/swagger/swagger.yaml:1018]: I don't think this should go in the swagger file.  This file is for the TA2-TA3 API, which doesn't involve ProbeOptions.  We just need the ProbeOptions (and Probe) model object for use with TA1.

File: swagger_server/itm/itm_scenario_session.py [hunk 2]
```diff
@@ -508,9 +508,50 @@ def lookup_probe_response(self, action: Action) -> ProbeResponse:
...
+                    if injury.location == action.parameters.location:
+                        casualty.remove(injury)
+                        break
+            
+            # if tagging a casualty then update the tag to the category parameter
+            if action.action_type == "TAG_CASUALTY":
+                 # getattr to account for partially specified action. If they don't tell us what to change the tag to keep it the same
+                tag = getattr(action.parameters, 'category', casualty.tag)
+                self.tag_casualty(self.session_id, casualty.id, tag)
+
```
Comment (nextcen-dgemoets) [inline swagger_server/itm/itm_scenario_session.py:535]: I'm not seeing the following from the ticket:
- Flip Casualty flags if necessary based on action (e.g., assessed , ~~tag~~)
- Flip Vitals flags if necessary based on action (e.g., conscious, responsive, breathing)

Comment (dereknop) [reply swagger_server/itm/itm_scenario_session.py:535]: I thought by calling tag_casualty the flag would be set. In that function casualty.tag gets set

Comment (dereknop) [reply swagger_server/itm/itm_scenario_session.py:535]: `for casualty in self.scenario.state.casualties:
            if casualty.id == casualty_id:
                casualty.tag = tag
                response = self.scenario.state
                self._add_history(
                    "Tag Casualty",
                    {"Session ID": self.session_id, "Casualty ID": casualty_id, "Tag": tag},
                    response.to_dict())
                return response`

File: swagger_server/itm/itm_scenario_session.py [hunk 3]
```diff
@@ -508,9 +508,50 @@ def lookup_probe_response(self, action: Action) -> ProbeResponse:
...
+            # I don't think updating vitals does anything here because the get_vitals and get heart rate funcs 
+            # just return what is already in the casualties vitals field. Probably not needed but was included in ticket
+            if action.action_type == "CHECK_ALL_VITALS":
+                vitals = self.get_vitals(self.session_id, casualty.id)
+                casualty.vitals.hrpmin = vitals.hrpmin
+                casualty.vitals.breathing = vitals.breathing
+                casualty.vitals.responsive = vitals.responsive
+                casualty.vitals.rr = vitals.rr
+                casualty.vitals.conscious = vitals.conscious
+                casualty._vitals.mm_hg = vitals.mm_hg
```
[the remark this answers is not in the document]
Comment (dereknop) [reply swagger_server/itm/itm_scenario_session.py:545]: Will do

File: swagger_server/itm/itm_scenario_session.py [hunk 4]
```diff
@@ -508,9 +508,50 @@ def lookup_probe_response(self, action: Action) -> ProbeResponse:
...
+
+            if action.action_type == "CHECK_PULSE":
+                casualty.vitals.hrpmin = self.get_heart_rate(self.session_id, casualty.id)
+            
+        else:
+            print("Error, casualty id of action not found")
```
[the remark this answers is not in the document]
Comment (dereknop) [reply swagger_server/itm/itm_scenario_session.py:551]: Yea that is placeholder

File: swagger_server/itm/itm_scenario_session.py (hunk 2 above)
Comment (nextcen-dgemoets) [reply swagger_server/itm/itm_scenario_session.py:535]: Yes, that's why I crossed out the word "tag" in my comment.  Everything else in those two bullets needed to be addressed, I believe  (assessed, conscious, responsive, breating).

**Pull request merged**